uint16 angle
float32 velocity